modules/videostab/src/stabilizer.cpp

```diff
@@ -391,6 +391,16 @@ void TwoPassStabilizer::runPrePassIfNecessary()
         {
             if (frameCount_ > 0)
             {
+                Mat feature_mask = Mat();
+                if (maskSource_) {
+                    Mat mask_frame = maskSource_->nextFrame();
+                    if (!mask_frame.empty())
+                    {
+                        feature_mask = Mat(mask_frame.size(), CV_8UC1);
+                        cv::cvtColor(mask_frame, feature_mask, cv::COLOR_BGR2GRAY);
```

Comment:
It is better to move such logic into "maskSource_" implementation.

which should produce binary images (0, 255) instead of grayscale.

Reply:
Any suggestions? `maskSource_` is of type `IFrameSource`, which is defined in opencv (not opencv_contrib). Do you want to introduce a new class inheriting from `IFrameSource`, which overwrites `nextFrame()`?

Reply:
Users should define their own class for mask generator, derived from `IFrameSource` interface (with inner conversions).
This class is not needed in OpenCV library itself (except probably tests).